### DISCUSSION TITLE
I want to understand shadowSocks

### DISCUSSION THREAD

**Discussion by Sjakelien** [shadowsocks/shadowsocks-org#248] [General]
Hi, I live the The Netherlands and my friend lives  in Nazi-Russia.
How would my friend benefit from shadowSocks, and how can I help him?
I have slightly more that average IT-knowledge and skills. E.g. I'm not scared of the command line tool. 
But what IS shadowSocks? Do I need to setup a server? Are there servers already? Are there compiled clients yet, that easily install on home pcs/mobile phones? etc, etc

**Comment by neteroster** [comment]
As stated on the official website, Shadowsocks is a tunneling proxy program designed primarily to help users bypass "network firewalls."

Here's a brief background introduction: As widely known, many websites, including Google and related services, cannot be directly accessed within China. This is because the firewall detects that the user's target domain or IP address is on a "blacklist" and blocks access through methods such as DNS pollution and packet interception. To bypass this restriction, we need an "intermediary," commonly referred to as a "server" in the context of Shadowsocks, as shown below:

User (Client) <= Firewall => Server <=> Google / etc.

By setting up a server outside the firewall, users can bypass the firewall's blacklist and successfully access blocked websites.

Originally, Shadowsocks was expected to be hard to be detected and distinguished from other traffic. However, subsequent experiments and discussions have shown that this may not be guaranteed. Nevertheless, using newer implementations can ensure the security of the data transmitted through the Shadowsocks tunnel.

In summary, Shadowsocks provides you with a lightweight, secure, and fast proxy service.

Regarding your questions:

* Do you need to set up a server? Yes, you need to configure both the server and client sides of Shadowsocks.
* Are there precompiled server / client versions available? Yes, Shadowsocks has many implementations, and you can find open-source GUI clients for [Windows](https://github.com/shadowsocks/shadowsocks-windows), [Linux](https://github.com/shadowsocks/shadowsocks-libev), [Android](https://github.com/shadowsocks/shadowsocks-android) and [macOS](https://github.com/shadowsocks/ShadowsocksX-NG). The commonly used server implementations are [C++](https://github.com/shadowsocks/shadowsocks-libev) and [Rust](https://github.com/shadowsocks/shadowsocks-rust) implementations. You can find detailed guide from repo pages.

**Reply by Sjakelien** [reply]
Thanks, that was very useful, and just the bit of information I need, for now.
To my understanding, I would set up a server outside Russia for my friend's client to connect more or less stealthily to any website outside the Great Russian Wall (which doesn't exist yet, but it's being worked on)

**Reply by neteroster** [reply]
That's right. Again, it should be noted that Shadowsocks could only ensure the content of your traffic is safe, and it may be possible to distinguish Shadowsocks traffic from other traffic (i.e. Your Internet provider may use some methods to detect shadowsocks traffic).

**Comment by igoldmax** [comment]
Nazi-Russia! Says the boy who likes Western noodles on his ears. Sjakelien, i'm sorry but you fool and you've been duped )))

**Reply by ivanstepanovftw** [reply]
Here's your [10 rubles](#)! Take it and go away from american internet now!

**Reply by ivanstepanovftw** [reply]
I think Microsoft should restrict access from ruzzan IP's to open source.
